classes/checkout/CheckoutAddressesStep.php

```diff
@@ -302,4 +304,23 @@ public function render(array $extraParams = array())
             $this->getTemplateParameters()
         );
     }
+    
+    public function setComplete($step_is_complete)
+    {
+        if ($step_is_complete &&
+                $this->context->cart->id_carrier == 0) {
+            $deliveryOptionSelected =
+                    $this->getCheckoutSession()->getSelectedDeliveryOption();
+            if ($deliveryOptionSelected) {
```

Comment:
You set the carrier in the address step, but how comes you already have the delivery option available?
Is it because you applied the voucher first? and it sets the `$this->getCheckoutSession()->getSelectedDeliveryOption()` option?

Reply:
I set the default address if it is available. The same method is being invoked on the next step, so I just speed up the process.